packages/react-scripts/config/webpack.config.dev.js

```diff
@@ -64,7 +83,7 @@ module.exports = {
     // containing code from all our entry points, and the Webpack runtime.
     filename: 'static/js/bundle.js',
     // In development, we always serve from the root. This makes config easier.
-    publicPath: '/'
+    publicPath: publicPath 
```

Comment:
This assumes people are going to serve both index.html and the js bundle at different hosts in dev just like they would do in production.

For most of my local dev I'm loading a react app into a rails view, (not using cra's generated html) during local dev and production. This makes that kind of thing possible without changing how it works for anyone not using $CDN_URL.